app/controllers/hashtags_controller.rb

```diff
@@ -19,11 +22,16 @@ def show
 
     # Try to resolve hashtags as categories first
     category_slugs_and_ids = all_slugs.map { |slug| [slug, Category.query_from_hashtag_slug(slug)&.id] }.to_h
-    categories_by_id = Category.secured(guardian).where(id: category_slugs_and_ids.values).map { |c| [c.id, c] }.to_h
+    category_ids_and_urls = Category
+      .secured(guardian)
+      .select(:id, :slug, :parent_category_id)
```

Comment:
Can you use `pluck` instead?

Reply:
`Category#url` uses the `parent_category` field.

Reply:
I see, might want to add a comment then.

Also, does this work with sub-sub-categories?

Reply:
Added the comment and yes, it does.